client/src/main/java/gui/utility/Command.java

```diff
@@ -0,0 +1,49 @@
+package gui.utility;
+
+import java.util.Objects;
+
+public class Command<T> {
+    private CommandType command;
+    private T value;
+
+    private Command(CommandType command, T value) {
+        this.command = command;
+        this.value = value;
+    }
+
+    public static <T> Command withType(CommandType command, T value) {
```

Comment:
method name - you provide here not only type, but also value. Consider builder pattern with two separate methods to create Command object or just rename it